src/transaction/log_replication_atomic.cpp

```diff
@@ -239,8 +239,14 @@ namespace cublog
 	    const LOG_REC_SCHEMA_MODIFICATION_LOCK log_rec =
 		    m_redo_context.m_reader.reinterpret_copy_and_add_align<LOG_REC_SCHEMA_MODIFICATION_LOCK> ();
 
+	    if (is_already_locked_for_ddl (header.trid, &log_rec.classoid))
+	      {
+		break;
+	      }
+
 	    acquire_lock_for_ddl (thread_entry, header.trid, &log_rec.classoid);
 
+	    bookkeep_classname (thread_entry, &log_rec.classoid);
```

Comment:
Nit. 
It needs a more specific name like
- `bookkeep_classname_for_ddl`,
- `bookkeep_classname_for_cache` or something describing its purpose.

Reply:
bookkeep_classname_for_ddl is better, I will change